R/utils.R

```diff
@@ -437,18 +442,187 @@ rot_x <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
+#' @export
+.rotxy.gg <- function(x, axis = "x", ...) {
+  if(axis=="x") {
+    x <- x + rot_x(...)
+  } else {
+    x <- x + rot_y(...)
+  }
+  x
+}
+#' @export
+.rotxy.patchwork <- function(x, axis = "x", ...) {
+  stopifnot(requireNamespace("patchwork", quietly = TRUE))
+  if(axis=="x") {
+    x <- x & rot_x(...)
+  } else {
+    x <- x & rot_y(...)
+  }
+  x
+}
+#' @export
+.rotxy.default <- function(x, ...) {
+  stop("invalid object; can only rotate gg or patchwork objects.")
+}
+
+#' Rotate axis tick marks in a list of plots
+#'
+#' Pass in a list of gg or patchwork objects and rotate tick marks on x or y
+#' axes.
+#'
+#' @param x a named list of gg or patchwork objects.
+#' @param at a character vector of list names to rotate.
+#' @param re a regular expression for selecting names to be used for `at`.
+#' @param axis which axis to rotate.
+#' @param ... additional arguments passed to [rot_x()] or [rot_y()].
+#'
+#' @details
+#' Note that all plots in the list need to be named.
+#'
+#' @seealso [rot_xy()], [rot_x()], [rot_y()].
+#'
+#' @examples
+#' data <- pmplots_data_id()
+#'
+#' co <- c("STUDYc", "CPc", "RF")
+#' etas <- paste0("ETA", 1:3)
+#'
+#' x <- eta_cat(data, x = co, y = etas)
+#' names(x)
+#'
+#' x <- rot_at(x, at = "ETA1vRF", angle = 35)
+#' x$ETA1vRF
+#'
+#' x <- rot_at(x, re = "RF", vertical = TRUE)
+#' x$ETA2vRF
+#' x$ETA3vRF
+#'
+#' @md
+#' @export
+rot_at <- function(x, at = names(x), re = NULL, axis = c("x", "y"), ...) {
+  if(!is.list(x) || is.ggplot(x) || inherits(x, "patchwork")) {
+    abort("`x` must be a list of gg or patchwork objects.")
+  }
+  if(!is_named(x)) abort("`x` must be named.")
+  axis <- match.arg(axis)
+  if(is.character(re)) {
+    where <- grep(re, names(x), perl = TRUE)
+  } else {
+    if(!is.character(at)) abort("`at` must be character.")
+    bad <- setdiff(at, names(x))
+    if(length(bad)) {
+      names(bad) <- rep("x", length(bad))
+      abort("requested names not found in `x`.", body = bad)
+    }
+    where <- which(names(x) %in% at)
+  }
+  if(!length(where)) {
+    warn("did not find any plots for axis rotation.")
+    return(x)
+  }
+  for(w in where) {
+    x[[w]] <- .rotxy(x[[w]], axis = axis, ...)
+  }
+  x
+}
+
+#' Rotate the x or y axes for gg or patchwork objects
+#'
+#' This is a wrapper function around [rot_at()], [rot_x()], and [rot_y()] that
+#' accepts gg objects, patchwork objects, or lists of gg or patchwork objects.
+#' Arguments to those wrapped functions can be passed through. See arguments
+#' to [rot_at()] as well as [rot_x()] and [rot_y()].
+#'
+#' @param x a gg object, patchwork object, or list of those objects.
+#' @param axis which axis to rotate (`x` or `y`).
+#' @param ... arguments passed to [rot_at()] (`at`, `re`) or [rot_x()] / [rot_y()]
+#' (`angle`, `vertical` `hjust`, `vjust`).
+#'
+#' @seealso [rot_at()], [rot_x()], [rot_y()].
+#'
+#' @examples
+#' library(patchwork)
+#'
+#' data <- pmplots_data_obs()
+#' id <- pmplots_data_id()
```

Comment:
The `id` assignment isn't used downstream.

Reply:
Removed.